commit 694fc7727f5e908a969275251414d7e523c74812 (refs/changes/64/4664/2)
Author: sjorge <sjorge@blackdot.be>
Date:   2018-08-11T17:53:54+02:00 (1 year, 2 months ago)
    
    OS-7116 bhyve brand should accept bios and uefi values for bootrom

diff --git a/usr/src/lib/brand/bhyve/zone/boot.c b/usr/src/lib/brand/bhyve/zone/boot.c
index e29616a748..f3f512e51d 100644
--- a/usr/src/lib/brand/bhyve/zone/boot.c
+++ b/usr/src/lib/brand/bhyve/zone/boot.c
@@ -38,7 +38,8 @@
 
 #define	ZH_MAXARGS		100
 
-#define	DEFAULT_BOOTROM		"/usr/share/bhyve/uefi-csm-rom.bin"
+#define	DEFAULT_BOOTROM		"/usr/share/bhyve/uefi-rom.bin"
+#define	DEFAULT_BOOTROM_CSM	"/usr/share/bhyve/uefi-csm-rom.bin"
 
 typedef enum {
 	PCI_SLOT_HOSTBRIDGE = 0,	/* Not used here, but reserved */
@@ -419,6 +420,11 @@ add_lpc(int *argc, char **argv)
 		}
 		if (i == bootrom_idx) {
 			found_bootrom = B_TRUE;
+			if (is_env_string("attr", lpcdevs[i], NULL, "bios")) {
+				strcpy(val, DEFAULT_BOOTROM_CSM);
+			} else if (is_env_string("attr", lpcdevs[i], NULL, "uefi")) {
+				strcpy(val, DEFAULT_BOOTROM);
+                        }
 		}
 		if (snprintf(conf, sizeof (conf), "%s,%s", lpcdevs[i], val) >=
 		    sizeof (conf)) {
@@ -434,7 +440,7 @@ add_lpc(int *argc, char **argv)
 
 	if (!found_bootrom) {
 		if (add_arg(argc, argv, "-l") != 0 ||
-		    add_arg(argc, argv, "bootrom," DEFAULT_BOOTROM) != 0) {
+		    add_arg(argc, argv, "bootrom," DEFAULT_BOOTROM_CSM) != 0) {
 			return (-1);
 		}
 	}
